src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureSelector.h

```diff
@@ -0,0 +1,265 @@
+
+    void setSegmentWindowLength(const Int segment_window_length);
+    Int getSegmentWindowLength() const;
+
+    void setSegmentStepLength(const Int segment_step_length);
+    Int getSegmentStepLength() const;
+
+    void setVariableType(const String& variable_type);
+    String getVariableType() const;
+
+    void setOptimalThreshold(const double optimal_threshold);
+    double getOptimalThreshold() const;
+
+    void setScoreWeights(const std::map<String, String>& score_weights);
+    std::map<String, String> getScoreWeights() const;
+
+protected:
+    /**
+      Add variable to the LP problem instantiated in `optimize()`
+
+      @param[in,out] problem LPWrapper object
+      @param[in] name Column name
+      @param[in] bounded Double bounded if true, otherwise Unbounded.
+      @param[in] obj Objective value
+
+      @return The variable's column index
+    */
+    Int addVariable_(
+      LPWrapper& problem,
+      const String& name,
+      const bool bounded = true,
+      const double obj = 1.0
+    ) const;
+
+    /**
+      Scoring method used by the optimizer. Metavalues to use are decided by
+      the `score_weights_` class' member.
+      The returned value is used in the LP problems' variables and contraints.
+
+      @param[in] feature Input feature
+
+      @return Computed score
+    */
+    double computeScore_(const Feature& feature) const;
+
+    /**
+      Add constraint to the LP problem instantiated in `optimize()`
+
+      @param[in,out] problem LPWrapper object
+      @param[in] indices LP matrix indices
+      @param[in] values LP matrix values
+      @param[in] name Row name
+      @param[in] lb Lower bound
+      @param[in] ub Upper bound
+      @param[in] param Row type
+    */
+    void addConstraint_(
+      LPWrapper& problem,
+      std::vector<Int> indices,
+      std::vector<double> values,
+      const String& name,
+      const double lb,
+      const double ub,
+      const LPWrapper::Type param
+    ) const;
+
+private:
+    Int    nn_threshold_            = 4;
+    bool   locality_weight_         = false;
+    bool   select_transition_group_ = true;
+    Int    segment_window_length_   = 8;
+    Int    segment_step_length_     = 4;
+    String variable_type_           = "continuous";
+    double optimal_threshold_       = 0.5;
+    std::map<String, String> score_weights_;
+
+    /**
+      Construct the target transition's or transition group's retention times that
+      will be used to score candidate features based on their deviation from the
+      relative distance between the target transition's or transition group's times
+
+      @param[in] features Input features
+      @param[out] time_to_name Pairs representing a mapping of retention times to transition names
+      @param[out] feature_name_map Transitions' names to their features objects
+    */
+    void constructTargTransList_(
+      const FeatureMap& features,
+      std::vector<std::pair<double, String>>& time_to_name,
```

Comment:
do these need to be sorted? I assume not...

Reply:
The `optimize` method makes use of two indices when accessing the elements of `time_to_name`. That is why I kept it as a `vector`.